keras_cv/models/backbones/detectron2/detectron2_backbone.py

```diff
@@ -0,0 +1,203 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+
+from keras_cv.backend import keras
+from keras_cv.layers.detectron2_layers import ViTDetPatchingAndEmbedding
+from keras_cv.layers.detectron2_layers import WindowedTransformerEncoder
+from keras_cv.models.backbones.backbone import Backbone
+from keras_cv.models.backbones.detectron2.detectron2_backbone_presets import (
+    backbone_presets,
+)
+from keras_cv.utils.python_utils import classproperty
+
+
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class ViTDetBackbone(Backbone):
```

Comment:
I assume that this backbone doesn't produce pyramid_level_outputs since it's a transformer architecture -- let's call this out in the docstring, and maybe even create an @property for self.pyramid_level_inputs which throws a nice NotImplementedError

Reply:
Addressed in [ac7f30e](https://github.com/keras-team/keras-cv/pull/1987/commits/ac7f30ee2cb0ee7569244cf67dd18a50ef805752)

Reply:
The backbone outputs the same shapes itself, but they do use a feature pyramid output: https://arxiv.org/pdf/2203.16527.pdf

TL:DR for the paper, the simple feature pyramid on the right turned out to be the most performant for them. 
<img width="755" alt="image" src="https://github.com/keras-team/keras-cv/assets/60978046/a3453f86-f96a-4f0a-8856-a3a8855f4b70">

Reply:
Hi @DavidLandup0, thanks for pointing out the paper! I didn't know the authors also proposed a FPN! I did look into the paper but don't know how the pyramid-level inputs would fit in the backbone here. Given this PR has already blown up a bit, I'd prefer to do this as a follow-up. Maybe you can take it up if you have time :)

Reply:
Doing this as a follow-up sgtm